packages/lisk-transactions/src/base_transaction.ts

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+// tslint:disable-next-line no-reference
+/// <reference path="../../../types/browserify-bignum/index.d.ts" />
+// See issue #912 for above reference
+
+import * as cryptography from '@liskhq/lisk-cryptography';
+import BigNum from 'browserify-bignum';
+import { BYTESIZES, TRANSACTION_TYPES } from './constants';
+import { TransactionError } from './errors';
+import {
+	Account,
+	StateReturn,
+	TransactionAsset,
+	TransactionJSON,
+	ValidateReturn,
+	VerifyReturn,
+} from './transaction_types';
+import {
+	checkBalance,
+	normalizeInput,
+	validateTransaction,
+	verifyTransaction,
+} from './utils';
+
+export abstract class BaseTransaction {
+	public readonly amount: BigNum;
+	public readonly fee: BigNum;
+	public readonly id: string;
+	public readonly recipientId: string;
+	public readonly recipientPublicKey: string;
+	public readonly senderId: string;
+	public readonly senderPublicKey: string;
+	public readonly signature?: string;
+	public readonly signatures?: ReadonlyArray<string> = [];
+	public readonly signSignature?: string;
+	public readonly timestamp: number;
+	public readonly type: number;
+	public readonly asset: TransactionAsset = {};
+	private applied = false;
+
+	public constructor(rawTransaction: TransactionJSON) {
+		normalizeInput(rawTransaction);
+		this.amount = new BigNum(rawTransaction.amount);
+		this.asset = rawTransaction.asset;
+		this.fee = new BigNum(rawTransaction.fee);
+		this.id = rawTransaction.id;
+		this.recipientId = rawTransaction.recipientId;
+		this.recipientPublicKey = rawTransaction.recipientPublicKey;
+		this.senderId = rawTransaction.senderId;
+		this.senderPublicKey = rawTransaction.senderPublicKey;
+		this.signature = rawTransaction.signature;
+		this.signatures = rawTransaction.signatures;
+		this.signSignature = rawTransaction.signSignature;
+		this.timestamp = rawTransaction.timestamp;
+		this.type = rawTransaction.type;
+	}
+
+	public abstract prepareTransaction(
```

Comment:
I think it's ok to remove this.
we just need to implement `public static create(xxx): TransactionJSON` function for each type of transaction i think